### PR TITLE
Added ability to set sun direction

### PR DESCRIPTION
**Pull request by prewettg**
I reworked the nesting of the layouts, I also changed some of the spacing to use MakeStretch() (a stretch element takes up as much space as it can get without exceeding the width of the layout).

I put the sliders and such in a VGrid, so that I could put the labels next to slider and have the widths be consistent. The only problem here is that VGrid is too simple to be able span columns, so we can't use one big VGrid because the section headers would mess up the spacing. This could give us inconsistent widths for column 1 (the labels). Fortunately, everything is going to the same text, so it happens to work.

The main reason for reorganizing is so that we can add spotlights in a future PR.

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/1548)

### CODE REVIEW

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- src/Open3D/Visualization/Rendering/Filament/FilamentScene.cpp  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5087459&bid=16737765)

**Comment by Xardvor** [comment]
Everything is ok

**Comment by Xardvor** [comment]
Hm, wait, I miss something

File: src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp [hunk 1]
```diff
@@ -186,6 +179,19 @@ class DrawTimeLabel : public gui::Label {
     gui::Window *window_;
 };
 
+class SmallButton : public gui::Button {
+    using Super = Button;
+
+public:
+    SmallButton(const char *title) : Button(title) {}
```
Comment (Xardvor) [inline src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp]: Could you make this constructor `explicit`?

Comment (prewettg) [reply src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp]: Good catch!

Comment (Xardvor) [reply src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp]: Thanks to Codacy))

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/Open3D/Visualization/Rendering/Filament/FilamentScene.cpp  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5087459&bid=16737765)

**Pull request merged**